### ISSUE TITLE
Reducing Memory Usage of Char-Level Text Generation by LSTMs

### ISSUE THREAD

**Issue by mvarga6** [keras-team/keras#9468]
First off, these examples are wonderful.  That being said, I have a question/improvement to the [char-level text generation example](https://github.com/keras-team/keras/blob/master/examples/lstm_text_generation.py).

I was trying this example out with a corpus of my own and kept running to MemoryError when allocating the training set and labels.  I noticed that both use a 1-hot encoding to indicate the activated character. 

**My question is**:  Why not bake this embedding into the model with an identity matrix before the first LSTM (basically just a character embedding layer that returns the 1-hot) and using a sparse categorical loss evaluation?  

This reduces the total in-memory usage by a factor of the number of unique characters.  Seemed to work for me when I tried it out.  

**Note**:  I was trying much longer input sequences (>100) which was why I was using a lot of memory.

Open to thoughts, issues, etc.

**Comment by ScottishFold007** [comment]
yeah，you're right!
